liana/method/sc/_liana_pipe.py

```diff
@@ -528,3 +535,151 @@ def _trimean(a, axis=0):
+    cluster_mean
+        pandas.core.series.Series ([num present LR pairs x 1] This is the average expression in each cluster. Column of lr_res dataframe. It considers all genes in that cluster. mean{X}^k, k=1,...,num clusters)
+    cluster_std
+        pandas.core.series.Series ([num present LR pairs x 1] This is the standard deviation expression in each cluster. Column of lr_res dataframe. It considers all genes in that cluster. std{X}^k, k=1,...,num clusters)
+    cluster_counts
+        pandas.core.series.Series ([num present LR pairs x 1] This is the gene average expression in each cluster. Column of lr_res dataframe. cardinality of each cluster. n_k, k=1,...,num clusters)
+
+    Returns
+    ----------
+    probability (pandas.core.series.Series [num present LR pairs x 1] This is the proportion that a normal distribution is less than the gene_mean.)
+    
+    """
+    probability = norm.cdf(gene_mean, loc=cluster_mean, scale = cluster_std/np.sqrt(cluster_counts))
+    # If gene expression in specific cluster is 0, then the gene score must be 0.
+    probability[gene_mean==0] = 0
+    
+    return probability
+
+def _complex_score(lr_res, adata):
+    """
+    Compute the ligand and receptor score for each ligand-receptor pair considering scSeqcomm method. The scores are 
+    related to the genes expressed in each cluster of cells. The score takes into account the protein complexes to which
+    different subunits of ligand (receptor) are part of. The score is the geometric mean of the scores of the subunits.
+
+    Parameters
+    ----------
+    lr_res
+        pd.DataFrame (Dataframe of results for each ligand-receptor pair)
+    adata
+        anndata.AnnData (Annotated data object. For scSeqComm must not be filtered by gene expression.)
+
+    Returns
+    ----------
+    lr_res (returns the ligand-receptor pair dataframe with several new columns with gene scores and cluster statistics.)
+    
+    """
+    lr_res = _cluster_mean(lr_res, adata)
+    lr_res = _cluster_sd(lr_res, adata)
+    lr_res = _cluster_counts(lr_res, adata)
+    
+    lr_res['ligand_score'] = _gene_score(lr_res['ligand_means'],
+                                 lr_res['source_cluster_mean'],
+                                 lr_res['source_cluster_std'],
+                                 lr_res['source_cluster_counts'])
+    lr_res['receptor_score'] = _gene_score(lr_res['receptor_means'],
+                                lr_res['target_cluster_mean'],
+                                lr_res['target_cluster_std'],
+                                lr_res['target_cluster_counts'])
+    
+    # if the ligand or the receptor is composed by different subunits of the same complex, we evaluate the 
+    # geometric mean of the subunit scores.
+    lr_res['ligand_score'] = lr_res.groupby(['ligand_complex','source'])['ligand_score'] \
```

Comment:
@BaldanMatt I cannot keep this part, but I do plan to enable LIANA to be ran with different ways of handling complexes in the future. Currently, it's the min of the two subunits.

Also, this will not work for complexes for which some subunits are not expressed.